androidApp/src/main/java/com/mbta/tid/mbta_app/android/stopDetails/StopDetailsViewModel.kt

```diff
@@ -559,6 +572,45 @@ fun stopDetailsManagedVM(
         )
     }
 
+    LaunchedEffect(
+        groupByDirection,
+        stopId,
+        globalResponse,
+        stopData,
+        filters,
+        alertData,
+        pinnedRoutes,
+        now
+    ) {
+        if (groupByDirection) {
+            val schedules = stopData?.schedules
+            withContext(Dispatchers.Default) {
+                viewModel.setRouteCardData(
+                    if (
+                        globalResponse != null &&
+                            stopId != null &&
+                            stopId == stopData?.stopId &&
+                            schedules != null &&
+                            stopData?.predictionsLoaded == true
+                    ) {
+                        RouteCardData.routeCardsForStopList(
+                            globalResponse.getStop(stopId)?.childStopIds.orEmpty(),
+                            globalResponse,
+                            sortByDistanceFrom = null,
+                            schedules,
+                            stopData?.predictionsByStop?.toPredictionsStreamDataResponse(),
+                            alertData,
+                            now,
+                            pinnedRoutes,
+                            // TODO do something about filtered stop details
+                            context = RouteCardData.Context.StopDetailsUnfiltered
```

Comment:
Since the context is an argument to `routeCardsForStopList` but we use the same data for unfiltered and filtered stop details, this is going to be a problem when we're ready to move filtered stop details to consuming the RouteCardData, but it's not a problem yet, so my inclination is to not worry about it.